In [22]:
#Import libraries
import pandas as pd
import numpy as np
import seaborn as sb
import matplotlib.pyplot as plt
from statsmodels.stats.outliers_influence import variance_inflation_factor as vif
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score

%matplotlib inline

In [23]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [25]:
import pandas as pd

#Read input data
train_data = pd.read_csv('/content/drive/MyDrive/ML/titanic-ml-project/data/processed/train.csv')
test_data = pd.read_csv('/content/drive/MyDrive/ML/titanic-ml-project/data/processed/test.csv')


In [26]:
# Define features and target variable
X_train = train_data.drop('Survived', axis=1)
Y_train = train_data['Survived']

X_test = test_data.drop('Survived', axis=1)
Y_test = test_data['Survived']

## HistGradientBoostingClassifier

In [27]:
from sklearn.ensemble import HistGradientBoostingClassifier

# Initialize and fit the model
model = HistGradientBoostingClassifier()
model.fit(X_train, Y_train)  # Assuming you have X_train and Y_train
Y_pred = model.predict(X_test)


In [28]:
diff_df = pd.DataFrame({'Actual' : Y_test, 'Predicted': Y_pred})
diff_df

Actual  Predicted
0         0          0
1         1          0
2         0          0
3         0          0
4         1          0
..      ...        ...
413       0          0
414       1          1
415       0          0
416       0          0
417       0          0

[418 rows x 2 columns]

In [29]:
from sklearn.metrics import confusion_matrix

conf_matrix = confusion_matrix(Y_test, Y_pred)
conf_matrix

array([[241,  25],
       [ 32, 120]])

 **Observations:**<br>
 **True Negatives (TN) = 241**<br>
The model correctly predicted 241 people as "Didnt Survived."<br>
 **False Positives (FP) = 25**<br>
The model incorrectly predicted 25 people as Survived when they actually didn't (Type I error).<br>
 **False Negatives (FN) = 32**<br>
The model incorrectly predicted 32 people as "Didnt Survived" when they actually did (Type II error).<br>
 **True Positives (TP) = 120**<br>
The model correctly predicted 120 people as Survived.<br>

In [30]:
from sklearn.metrics import classification_report

target_names = ['Didnt Survived', 'Survived']
print(classification_report(Y_test, Y_pred, target_names=target_names))

                precision    recall  f1-score   support

Didnt Survived       0.88      0.91      0.89       266
      Survived       0.83      0.79      0.81       152

      accuracy                           0.86       418
     macro avg       0.86      0.85      0.85       418
  weighted avg       0.86      0.86      0.86       418

